runtime/jcl/common/mgmtthread.c

```diff
@@ -1120,18 +1111,19 @@ handlerMonitorWaited(J9HookInterface** hook, UDATA eventNum, void* eventData, vo
  * @retval JNI_ENOMEM OutOfMemoryError is set
  * @retval JNI_ERR FindClass() or GetMethodID() failed, an exception is set
  */
-static jint
-initIDCache(JNIEnv *env)
+void JNICALL
+Java_openj9_management_internal_IDCacheInitializer_initIDCache (JNIEnv *env, jclass jcls)
 {
 	jclass oom;
 	jclass cls;
 	jclass gcls;
 	jmethodID mid;
-	jint err = JNI_OK;
+	IDATA err = JNI_OK;
+	J9VMThread *currentThread = (J9VMThread *)env;
 	
 	/* isNativeMethod is the last cache member to be set */
 	if (JCL_CACHE_GET(env, MID_java_lang_StackTraceElement_isNativeMethod) != NULL) 
-		return JNI_OK;
+		return;
```

Comment:
IIUC, the test is to detect a second call (potentially from another caller) and avoid it as early as possible. So it is still necessary if we want to avoid duplication of efforts?

Reply:
If you want to keep the test, it should be an assertion, or throw something like IllegalStateException. My preference is to simply remove it.

Reply:
Class initialization ensures the single invocation of this code - the test should be removed.

Reply:
sure, got it, removed.